src/components/SVGViewer/SVGViewer.tsx

```diff
@@ -476,31 +479,39 @@ export class SVGViewer extends Component<SvgViewerProps, SvgViewerState> {
       className: string;
     }[]
   ) => {
-    const gElements = this.svg.querySelectorAll('.metadata-container');
+    const metadataContainers = this.svg.querySelectorAll('.metadata-container');
+    metadataContainers.forEach(metaContainer => {
+      // reset classes in case this method is being called more than once for the same document
+      // avoid resetting all the classes, as they may contain currentAsset class that affects zoom
+      metadataClassesAndConditions.forEach(({ className }) => {
+        metaContainer.classList.remove(className);
+      });
 
-    gElements.forEach(metadataContainer => {
       const metadataElements = Array.from(
-        metadataContainer.querySelectorAll('metadata')
+        metaContainer.querySelectorAll('metadata')
       );
 
+      // for the sake of performance we need to execute conditions as little as possible, to achieve it:
+      // - we don't execute condition for a metadata el if condition was met before
+      // - we don't iterate metadata anymore if all conditions are met
       const unmetConditions = [...metadataClassesAndConditions];
 
-      // reset classes in case this method is being called more than once for the same document
-      // avoid resetting all the classses, as they may contain currentAsset class that affects zoom
-      unmetConditions.forEach(({ className }) => {
-        metadataContainer.classList.remove(className);
-      });
-
       for (const metadataElement of metadataElements) {
         if (!unmetConditions.length) {
           // metadataContainer satisfies to all passed conditions, nothing left to look for
           break;
         }
-        for (let k = 0; k < unmetConditions.length - 1; k++) {
-          const { condition, className } = unmetConditions[k];
+
+        let unmetConditionIndex = 0;
+        while (unmetConditionIndex < unmetConditions.length) {
+          const { condition, className } = unmetConditions[unmetConditionIndex];
           if (condition(metadataElement)) {
-            metadataContainer.classList.add(className);
-            unmetConditions.splice(k, 1);
+            metaContainer.classList.add(className);
+            unmetConditions.splice(unmetConditionIndex, 1);
```

Comment:
Just wondering, but why are elements from the `unmetConditions` array being removed here?

Reply:
I kinda tried to explain it [here](https://github.com/cognitedata/gearbox.js/pull/732/files#diff-23b593734827286577a091e3c2f550964f5853adf83e063e3cea92a7a6b6a701R494-R496), but 2 reasons:

1 - to exit early and not iterate through the other metadata elements when parent already got the class because that condition was met
2 - to avoid checking the same condition when it already returned true for one of the metadata elements. We only need one medata tag to satisfy condition to apply passed class to the parent `g` tag that has `.metadata-container` class.

Reply:
```
unmetConditions
  .filter(({ condition }) => 
   // check if any metadataElement satisifies condition, one is enough
    metadataElements.some(metadataElement => condition(metadataElement))
 ).forEach(({ className}) => metaContainer.classList.add(className))
```

Would this do the trick as well?

Reply:
@Robin-Hoodie actually it will, and I like your variant readability way better than the current one and it even passes the tests. I regret merging now :D

Reply:
@maksnester I can open a PR to refactor this if you aren't already doing that?

Reply:
There is a tiny difference though. The current option tries to exit as early as possible from iteration through `metadataElements` while your suggestion always reiterates `metadataElements.some(condition)` to find one that matches the condition. 

Despite `metadataElements` number is normally noticeable bigger (still tens to hundreds though) than the number of passed conditions, I don't think it will give any noticeable performance degradation if we apply your suggested option. 

However, I'm a bit reluctant to apply it knowing that it must be (likely insignificantly but) slower on actual documents 🙄 

but readability is way better though...

Reply:
my point was that currently we go once or less over metadata elements, while in your example you iterate over them (using ‘some’ which can return early if there is a match for condition) for each condition item. 

But as I said I don’t think it will degrade performance much so we can go for it for better readability.